brian2/equations/equations.py

```diff
@@ -223,17 +226,27 @@ def dimensions_and_type_from_string(unit_string):
     ValueError
         If the string cannot be evaluated to a unit.
     '''
+    # Lazy import to avoid circular dependency
```

Comment:
Could we think about refactoring this stuff to the units code? Even though it's kind of specific to equations, it feels odd to have it here.